lua/plenary/nvim/init.lua

```diff
@@ -0,0 +1,34 @@
+
+local lookups = {
+  autocmd = "autocmds",
+  augroup = "autocmds",
+  keymap = "keymap",
+}
+
+local groups = {
+  autocmds = { major = 0, minor = 7 },
+  keymap = { major = 0, minor = 7 },
+}
+
+local version = vim.version()
+
+return setmetatable({}, {
+  __index = function(t, k)
+    local group = lookups[k]
+    if not group then
+      return
+    end
+
+    local info = groups[group]
+
+    local needs_compat =
```

Comment:
could maybe just check `if vim.api.nvim_create_autocmd` or something like that if you wanted.

Reply:
I just rewrote this whole PR to be a general compat module and I originally did just this. Even though core is against adding features flags, I guess if we can detect then we should.